### PR TITLE
setup connection to database with migrations and struct generation

### PR DESCRIPTION
**Pull request by cchirag**
The PR uses SQLC with PGX, Atlas, PostgreSQL to handle the state of the application.
SQLC: Used to transpile the SQL queries to go types and methods
PGX: SQL driver
Atlas: Migration tool
PostgreSQL: Database

The PR comes with an initial migration of all the basic tables and generates structs

### CODE REVIEW

File: .database/query.sql
Comment (chiragchan) [inline .database/query.sql:1]: Creation of app user should also create user preferences, add a query for insert into preferences table

File: Makefile
Comment (chiragchan) [inline Makefile:1]: Commands look ambiguous, without seeing that actual Makefile. Look for other alternatives for make.

File: .database/query.sql
Comment (chiragchan) [reply .database/query.sql:1]: Done

Comment (cchirag) [reply .database/query.sql:1]: Done

File: Makefile
Comment (cchirag) [reply Makefile:1]: Will be updated in another PR. Reviewing multiple other alternatives for makefile.

**Review by chiragchan — APPROVED** [review]
Can merge the branch. To be noted that an alternative to Makefile is to be found and used.

**Pull request merged**